### PR TITLE
feat: ✨ add project level manual scan [ROAD-1239]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

This PR adds a command to scan a workspace folder. 

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: application/server/execute_command.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ import (
 )
 
 func ExecuteCommandHandler(srv *jrpc2.Server) jrpc2.Handler {
-	return handler.New(func(ctx context.Context, params sglsp.ExecuteCommandParams) (interface{}, error) {
+	return handler.New(func(ctx context.Context, params sglsp.ExecuteCommandParams) (any, error) {
```
Comment (asaf92) [inline application/server/execute_command.go:38]: Nice

File: domain/ide/workspace/folder.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func (f *Folder) filterCachedDiagnostics() (fileIssues map[string][]snyk.Issue)
 
 func filterIssues(issues []snyk.Issue, supportedIssueTypes map[product.FilterableIssueType]bool) []snyk.Issue {
 	logger := log.With().Str("method", "filterIssues").Logger()
-	filteredIssues := []snyk.Issue{}
+	filteredIssues := make([]snyk.Issue, 0)
```
Comment (asaf92) [inline domain/ide/workspace/folder.go:197]: Why?

File: internal/uri/uri_util.go [hunk 3]
```diff
@@ -34,7 +34,9 @@ const eclipseWorkspaceFolderScheme = "file:"
 var rangeFragmentRegexp = regexp.MustCompile(`^(.+)://((.*)@)?(.+?)(:(\d*))?/?((.*)\?)?((.*)#)L?(\d+)(?:,(\d+))?(-L?(\d+)(?:,(\d+))?)?`)
 
 func FolderContains(folderPath string, path string) bool {
-	return strings.HasPrefix(path, folderPath)
+	log.Debug().Str("folderPath", folderPath).Str("path", path).Msg("FolderContains")
```
Comment (asaf92) [inline internal/uri/uri_util.go:37]: This is wrong, I added a failing test to demonstrate

**Review by asaf92 — CHANGES_REQUESTED** [review]
Trimming the suffix leads to wrong results in FolderContains, check the test I added

File: domain/ide/workspace/folder.go (hunk 2 above)
Comment (bastiandoetsch) [reply domain/ide/workspace/folder.go:197]: Goland likes it better that way - and I don't have any clue why.

File: internal/uri/uri_util.go (hunk 3 above)
Comment (bastiandoetsch) [reply internal/uri/uri_util.go:37]: Yeah, the folder must be normalized, but the file system path separator included in the contains check. Good catch!

Comment (bastiandoetsch) [reply internal/uri/uri_util.go:37]: Alternatively, we must save all folder paths with a trailing slash, which may be the better solution.

Comment (bastiandoetsch) [reply internal/uri/uri_util.go:42]: Please have a look now - have added a few more test scenarios and adjusted the function

Comment (asaf92) [reply internal/uri/uri_util.go:42]: Looks OK
I'd probably normalize both paths with filepath.Clean just to make it more robust

**Review by asaf92 — APPROVED** [review]

File: internal/uri/uri_util.go (hunk 3 above)
Comment (bastiandoetsch) [reply internal/uri/uri_util.go:42]: I'll add that and then merge
=

**Pull request merged**